network/network_linux.go

```diff
@@ -187,6 +254,182 @@ func (nm *networkManager) deleteBridgeRules(extIf *externalInterface) {
 	ebtables.SetSnatForInterface(extIf.Name, extIf.MacAddress, ebtables.Delete)
 }
 
+func (nm *networkManager) createOVSBridge(bridgeName string) error {
+	log.Printf("[net] Creating OVS Bridge %v", bridgeName)
+
+	ovsCreateCmd := fmt.Sprintf("ovs-vsctl add-br %s", bridgeName)
+	_, err := common.ExecuteShellCommand(ovsCreateCmd)
+	if err != nil {
+		log.Printf("[net] Error while creating OVS bridge %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func (nm *networkManager) deleteOVSBridge(bridgeName string) error {
+	log.Printf("[net] Deleting OVS Bridge %v", bridgeName)
+
+	ovsCreateCmd := fmt.Sprintf("ovs-vsctl del-br %s", bridgeName)
+	_, err := common.ExecuteShellCommand(ovsCreateCmd)
+	if err != nil {
+		log.Printf("[net] Error while deleting OVS bridge %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func setOVSMaster(hostIfName string, bridgeName string) error {
+	cmd := fmt.Sprintf("ovs-vsctl add-port %s %s", bridgeName, hostIfName)
+	_, err := common.ExecuteShellCommand(cmd)
+	if err != nil {
+		log.Printf("[net] Error while setting OVS as master to primary interface %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func (nm *networkManager) createOVSNetwork(extIf *externalInterface, nwInfo *NetworkInfo) error {
+	var err error
+
+	log.Printf("[net] Connecting interface %v.", extIf.Name)
+	defer func() { log.Printf("[net] Connecting interface %v completed with err:%v.", extIf.Name, err) }()
+
+	// Check whether this interface is already connected.
+	if extIf.BridgeName != "" {
+		log.Printf("[net] Interface is already connected to bridge %v.", extIf.BridgeName)
+		return nil
+	}
+
+	// Find the external interface.
+	hostIf, err := net.InterfaceByName(extIf.Name)
+	if err != nil {
+		return err
+	}
+
+	// If a bridge name is not specified, generate one based on the external interface index.
+	bridgeName := nwInfo.BridgeName
```

Comment:
We can do _if bridgeName := nwInfo.BridgeName; bridgeName != ""{}_

Reply:
bridgename will be used elsewhere in that function. so can't do this